src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java

```diff
@@ -0,0 +1,379 @@
+        fields.add(memberTypes);
+        final StructField memberRoles = DataTypes.createStructField("memberRoles",
+                DataTypes.createArrayType(DataTypes.StringType), true);
+        fields.add(memberRoles);
+        final StructField tags = DataTypes.createStructField("tags",
+                DataTypes.createMapType(DataTypes.StringType, DataTypes.StringType), true);
+        fields.add(tags);
+        final StructField relations = DataTypes.createStructField("relations",
+                DataTypes.createArrayType(DataTypes.StringType), true);
+        fields.add(relations);
+        final StructType schema = DataTypes.createStructType(fields);
+
+        // Convert entities of atlas to Rows
+        final List<Row> rows = new ArrayList<>();
+        atlasRDD.foreach(atlas ->
+        {
+            atlas.relations().forEach(relation ->
+            {
+                final long nodeIdentifier = relation.getIdentifier();
+                final String idString = String.valueOf(nodeIdentifier);
+                final String wktGeometry = relation.toWkt();
+                final String[] relationMemberIdentifiers = relation.members().stream().sorted()
+                        .map(member -> String.valueOf(member.getEntity().getIdentifier()))
+                        .toArray(String[]::new);
+                final String[] relationMemberRoles = relation.members().stream().sorted()
+                        .map(member -> String.valueOf(member.getRole())).toArray(String[]::new);
+                final String[] relationMemberTypes = relation.members().stream().sorted()
+                        .map(member -> String.valueOf(
+                                typeValueToString(member.getEntity().getType().getValue())))
+                        .toArray(String[]::new);
+                final String[] relationArray = relation.relations().stream()
+                        .map(parentRrelation -> String.valueOf(parentRrelation.getIdentifier()))
+                        .toArray(String[]::new);
+                rows.add(RowFactory.create(idString, wktGeometry, relationMemberIdentifiers,
+                        relationMemberTypes, relationMemberRoles,
+                        JavaConversions.mapAsScalaMap(relation.getTags()), relationArray));
+            });
+        });
+        final JavaSparkContext sparkContext = JavaSparkContext
+                .fromSparkContext(spark.sparkContext());
+        final JavaRDD<Row> rowRDD = sparkContext.parallelize(rows);
+
+        // Create dataframe from rowRDD
+        return spark.createDataFrame(rowRDD, schema);
+    }
+
+    private static String typeValueToString(final int value)
+    {
+        final int line = 3;
+        final int point = 4;
+        final int relation = 5;
+        switch (value)
+        {
+            case 0:
+                return "NODE";
+            case 1:
+                return "EDGE";
+            case 2:
```

Comment:
Aah checkstyle! Mabe make it consistent with the ones below?

Reply:
yes, definitely a checkstyle-ism :)